libpdbg/device.c

```diff
@@ -48,11 +48,118 @@ static const char *take_name(const char *name)
 	return name;
 }
 
+static const char *get_unitname(const struct pdbg_target *node)
+{
+	const char *c = strchr(node->dn_name, '@');
+
+	if (!c)
+		return NULL;
+
+	return c + 1;
+}
+
+/*If a string contains alphanumeric characters then it extracts 
+	the number from that string (from the end only) and 
+	returns the extracted number and the chopped off string*/
+static unsigned int strip_num_at_the_end(char *str) 
+{
+	unsigned int only_number;
+	int len = strlen(str);
+	char new_num_str[len];
+
+	int i;
+	/*Traverse from the end of the string looking for alphabet after the digit */
+	for(i = (len - 1); i >= 0; i--) {
+		if(!isdigit(str[i])) {
+		/* This is the first character from the end. So 
+		copy the number starting from the next index */
+		strcpy(new_num_str, &str[i+1]);
+		break;
+		}
+	}
+	str[i+1] = '\0';
+	only_number = atoi(new_num_str);
+	return only_number;
+}
+
+/*Compare the pdbg_targets according to their name. Initially checks where it falls 
+as per alphabetical order. Then from there use the number at the end of the name to 
+sort numerically within that group
+e.g. perv3 shall be placed after omi0 and before perv10 */
+static int dt_cmp_subnodes(const struct pdbg_target *a, const struct pdbg_target *b)
```

Comment:
this too do not change the location of the function or else it will show as new function, if required you can do forward declaration.

Reply:
Done